miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/BulkTableBackend.java

```diff
@@ -0,0 +1,76 @@
+package uk.ac.bbsrc.tgac.miso.webapp.util;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.ui.ModelMap;
+import org.springframework.web.servlet.ModelAndView;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+public abstract class BulkTableBackend<Model, Dto, ParentModel> {
+  private final String name;
+  private final String parentName;
+  private final String targetType;
+  public BulkTableBackend(String name, String parentName, String targetType) {
+    super();
+    this.name = name;
+    this.parentName = parentName;
+    this.targetType = targetType;
+  }
+
+  protected abstract Dto asDto(Model model);
+
+  public ModelAndView edit(String idString, ModelMap model) throws IOException {
+    List<Long> ids = parseIds(idString);
+    List<Dto> dtos = new ArrayList<>();
+    for (Model item : load(ids)) {
+      dtos.add(asDto(item));
+    }
+    return prepare(idString, model, false, "Edit " + name, dtos);
+  }
+
+  protected abstract Class<? extends Dto> getDtoClass();
+
+  protected abstract Iterable<Model> load(List<Long> ids) throws IOException;
+
+  protected abstract Iterable<ParentModel> loadParents(List<Long> ids) throws IOException;
+
+  protected abstract Dto packageDto(ParentModel item);
+
+  private List<Long> parseIds(String idString) {
+    String[] split = idString.split(",");
+    List<Long> ids = new ArrayList<>();
+    for (int i = 0; i < split.length; i++) {
+      ids.add(Long.parseLong(split[i]));
+    }
+    return ids;
+  }
+
+  private ModelAndView prepare(String idString, ModelMap model, boolean create, String title, List<Dto> dtos) throws IOException {
+    ObjectMapper mapper = new ObjectMapper();
+    ObjectNode config = mapper.createObjectNode();
+    writeConfiguration(mapper, config);
+    model.put("title", title);
+    model.put("config", mapper.writeValueAsString(config));
+    model.put("targetType", "HotTarget." + targetType);
+    model.put("create", create);
+    model.put("input",
+        mapper.writerFor(mapper.getTypeFactory().constructCollectionType(List.class, getDtoClass())).writeValueAsString(dtos));
+    model.put("method", "Propagate");
+    return new ModelAndView("/pages/handsontables.jsp", model);
+  }
+
+  public ModelAndView propagate(String idString, ModelMap model) throws IOException {
+    List<Long> ids = parseIds(idString);
+    List<Dto> dtos = new ArrayList<>();
+    for (ParentModel item : loadParents(ids)) {
+      dtos.add(packageDto(item));
+    }
+    return prepare(idString, model, true, "Create " + name + " from " + parentName, dtos);
+  }
+
+  protected abstract void writeConfiguration(ObjectMapper mapper, ObjectNode config);
```

Comment:
Yes. It's going to be needed in the sample version where we want to ship around the sample class.